### PR TITLE
Add polygon helper

### PR DESCRIPTION
**Pull request by jonathanlurie**
[RD-135](https://maptiler.atlassian.net/browse/RD-135)  

Adding a `Map` method to easily add styled polygons.
Examples to be found at:
- demos/polygon_switzerland.html
- demos/polygon_switzerland2.html

[RD-135]: https://maptiler.atlassian.net/browse/RD-135?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -13,7 +14,7 @@ class SdkConfig extends EventEmitter {
   /**
    * The primary language. By default, the language of the web browser is used.
    */
-  primaryLanguage: LanguageString = Language.AUTO;
+  primaryLanguage: LanguageString = defaults.primaryLanguage;
```
Comment (CraigglesO) [inline src/config.ts:17]: When you changed this, on line2 the import for Language is now unused.

File: src/Map.ts [hunk 2]
```diff
@@ -595,13 +600,47 @@ export class Map extends maplibregl.Map {
       const strMoreInfoRegex = /^(.*)({\s*name\s*(:\s*(\S*))?\s*})(.*)$/;
 
       const langStr = language ? `name:${language}` : "name"; // to handle local lang
-      const replacer = [
+      let replacer = [
         "case",
         ["has", langStr],
         ["get", langStr],
         ["get", "name"],
       ];
```
Comment (CraigglesO) [inline src/Map.ts:609]: The replaceer section can be replaced with `ExpressionSpecification` (a type from maplibre-gl):

```ts
let replacer: ExpressionSpecification = [
        "case",
        ["has", langStr],
        ["get", langStr],
        ["get", "name"],
      ];

      if (language == Language.VISITOR) {
        const firstLang = `name:${getBrowserLanguage()}`;
        replacer = [
          "case",
          ["all", ["has", firstLang], ["has", "name"]],

          [
            "case",
            ["==", ["get", firstLang], ["get", "name"]],
            ["get", "name"],
            ["concat", ["get", firstLang], "\n", ["get", "name"]],
          ],
          ["get", "name"],
        ];
      }
      ...
```

then you can avoid all the `@ts-ignore`

File: src/Map.ts [hunk 3]
```diff
@@ -1424,6 +1474,213 @@ export class Map extends maplibregl.Map {
...
+      if (options.outline === true) {
+        let computedOutlineOffset:
+          | DataDrivenPropertyValueSpecification<number>
+          | number;
+
+        if (outlinePosition === "inside") {
+          if (typeof outlineWidth === "number") {
+            computedOutlineOffset = 0.5 * outlineWidth;
+          } else {
+            // @ts-ignore
```
Comment (CraigglesO) [inline src/Map.ts:1578]:
```suggestion

```
For me this comment isn't necessary

File: src/Map.ts [hunk 4]
```diff
@@ -1424,6 +1474,213 @@ export class Map extends maplibregl.Map {
...
 
-    return retunedInfo;
+    return returnedInfo;
+  }
+
+  /**
+   * Add a polygon with styling options.
+   */
+  addPolygon(
+    // this Feature collection is expected to contain on LineStrings and MultilLinestrings
```
Comment (CraigglesO) [inline src/Map.ts:1484]:
```suggestion
    // this Feature collection is expected to contain on LineStrings and MultiLinestrings
```

lint

I also really don't know what the correct one is here:
* FeatureCollection
* Feature Collection
* Feature collection

Up to you.

File: src/Map.ts [hunk 5]
```diff
@@ -1424,6 +1474,213 @@ export class Map extends maplibregl.Map {
...
+    const fillColor = options.fillColor ?? getRandomColor();
+    const fillOpacity = options.fillOpacity ?? 1;
+    const outlinePosition = options.outlinePosition ?? "center";
+    const pattern = options.pattern ?? null;
+
+    if (typeof outlineDashArray === "string") {
+      outlineDashArray = dashArrayMaker(outlineDashArray);
+    }
+
+    console.log("outlineDashArray", outlineDashArray);
```
Comment (CraigglesO) [inline src/Map.ts:1540]:
```suggestion
```

I'm assuming this was just left on accident.

File: src/Map.ts [hunk 6]
```diff
@@ -1424,6 +1474,213 @@ export class Map extends maplibregl.Map {
...
+        if (outlinePosition === "inside") {
+          if (typeof outlineWidth === "number") {
+            computedOutlineOffset = 0.5 * outlineWidth;
+          } else {
+            // @ts-ignore
+            computedOutlineOffset = rampedOptionsToLineLayerPaintSpec(
+              outlineWidth.map((el) => ({
+                zoom: el.zoom,
+                value: 0.5 * el.value,
+              })),
```
Comment (CraigglesO) [inline src/Map.ts:1583]:
```suggestion
              outlineWidth.map(({ zoom, value }) => ({
                zoom,
                value: 0.5 * value,
              })),
```
nit

File: src/Map.ts (hunk 4 above)
Comment (jonathanlurie) [reply src/Map.ts:1484]: Yeah i think my `Feature collection` is mix between the type name `FeatureCollection` and the English of it being a `feature collection`. But the truth is, it actaully does not have to be of type "FeatuerCollection"

**Pull request merged**